src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -99,6 +100,19 @@ internal static SafeSslHandle AllocateSslContext(SslProtocols protocols, SafeX50
                     context.Dispose();
                     throw CreateSslException(SR.net_allocate_ssl_context_failed);
                 }
+
+                if (certHandle != null && certHandle.IsInvalid)
```

Comment:
Really it just means that we haven't passed any null-but-invalid certhandles in; and with the inversion it just did precisely what it did before this change.

As I rambled about in the issue, I'm not really sure what a good way is to do automated testing on this.  Our client side makes up for deficient servers, so for best results we'd need the client to be as dumb as possible and just show the raw session contents (like `openssl s_client -showcerts`)... and it requires a cert chain depth of 3 or more.